### ISSUE TITLE
Modeling search contains only one peptide against thousands spectra?

### ISSUE THREAD

**Issue by cctsou** [percolator/percolator#218]
Hi,

I have an application where I have a lot of spectra and I want to find correct hits for only one peptide.
I tried percolator and got Segmentation fault exception. The following is the output from the execution.
It says "Tossing out "redundant" PSMs keeping only the best scoring PSM for each unique peptide", does that mean it doesn't not work for my case since I have only one peptide?

Thanks a lot,

> Percolator version 3.01, Build Date Nov 16 2016 14:58:22
> Copyright (c) 2006-9 University of Washington. All rights reserved.
> Written by Lukas Käll (lukall@u.washington.edu) in the
> Department of Genome Sciences at the University of Washington.
> Issued command:
> percolator test_FT.tsv
> Started Tue Jul  3 21:47:07 2018
> Hyperparameters: selectionFdr=0.01, Cpos=0, Cneg=0, maxNiter=10
> Reading tab-delimited input from datafile test_FT.tsv
> 
> WARNING: Tab delimited input does not contain ScanNr column,
>          scan numbers will be assigned automatically.
> 
> Features:
> DotProduct uRT_Score MassErrorScore
> Found 28106 PSMs
> Concatenated search input detected, skipping both target-decoy competition and mix-max.
> Train/test set contains 13786 positives and 14320 negatives, size ratio=0.962709 and pi0=1
> Selecting Cpos by cross-validation.
> Selecting Cneg by cross-validation.
> Split 1:        Selected feature 1 as initial search direction. Could separate 3925 training set positives in that direction.
> Split 2:        Selected feature 1 as initial search direction. Could separate 4014 training set positives in that direction.
> Split 3:        Selected feature 1 as initial search direction. Could separate 3918 training set positives in that direction.
> Found 5933 test set positives with q<0.01 in initial direction
> Reading in data and feature calculation took 0.109933 cpu seconds or 0 seconds wall clock time.
> ---Training with Cpos selected by cross validation, Cneg selected by cross validation, fdr=0.01
> Iteration 1:    Estimated 6221 PSMs with q<0.01
> Iteration 2:    Estimated 6222 PSMs with q<0.01
> Iteration 3:    Estimated 6222 PSMs with q<0.01
> Iteration 4:    Estimated 6224 PSMs with q<0.01
> Iteration 5:    Estimated 6223 PSMs with q<0.01
> Iteration 6:    Estimated 6223 PSMs with q<0.01
> Iteration 7:    Estimated 6224 PSMs with q<0.01
> Iteration 8:    Estimated 6223 PSMs with q<0.01
> Iteration 9:    Estimated 6223 PSMs with q<0.01
> Iteration 10:   Estimated 6223 PSMs with q<0.01
> Learned normalized SVM weights for the 3 cross-validation splits:
>  Split1  Split2  Split3 FeatureName
>  3.8185  4.2111  3.0201 DotProduct
> -0.0955 -0.2181 -0.1006 uRT_Score
>  1.2206  1.0622  0.8943 MassErrorScore
> -2.0184 -1.7676 -1.5600 m0
> Found 6196 test set PSMs with q<0.01.
> Tossing out "redundant" PSMs keeping only the best scoring PSM for each unique peptide.
> Calculating q values.
> Final list yields 0 target peptides with q<0.01.
> Calculating posterior error probabilities (PEPs).
> Log transforming all scores prior to PEP calculation
> Segmentation fault (core dumped)

**Comment by MatthewThe** [comment]
I think you might want PSM-level statistics instead of the default, which is peptide-level statistics (which probably fails because not enough peptides are available for PEP estimation). If you add the `-U` flag you will get PSM-level statistics.

**Comment by cctsou** [comment]
It worked, thanks a lot.
